src/qibolab/instruments/zhinst.py

```diff
@@ -431,18 +434,29 @@ def register_readout_line(self, qubit, intermediate_frequency):
         self.signal_map[f"acquire{q}"] = self.device_setup.logical_signal_groups[f"q{q}"].logical_signals[
             "acquire_line"
         ]
+
+        if qubit.kernel_path:
+            self.kernels[q] = qubit.kernel_path
+
+        oscillator = lo.Oscillator(
+            frequency=intermediate_frequency,
+            modulation_type=lo.ModulationType.SOFTWARE,
+        )
+        threshold = None
+
+        if options.acquisition_type == AcquisitionType.DISCRIMINATION:
+            if self.kernels[q].is_file():
+                # Kernels don't work with the software modulation on the acquire signal
+                oscillator = None
+            elif not self.kernels[q].is_file():
```

Comment:
```suggestion
            else:
```

:P

Reply:
Is it better to use else in most cases or using something like this to make it clearer is too much ?